### DISCUSSION TITLE
Supporting chrono after breaking sqlx change

### DISCUSSION THREAD

**Discussion by vinnymeller** [maxcountryman/tower-sessions#211] [Ideas]
Existing issue in sqlx repo: https://github.com/launchbadge/sqlx/issues/3412#issuecomment-2277377597

tldr: sqlx made a breaking change - https://github.com/launchbadge/sqlx/pull/2697 - where the `sqlx::types::chrono` types only get enabled if `time` is disabled

this crate and the `tower-sessions-stores` crates assume `time::OffsetDateTime` as the datetime type: https://github.com/maxcountryman/tower-sessions/blob/d7b7e1398b75e2bb4cea266824518943db2b2732/tower-sessions-core/src/session.rs#L569

I haven't had time to think about a nice solution yet but hadn't seen this posted anywhere here yet so thought I'd start the discussion. Currently it's not a great experience to work around this (described here https://github.com/launchbadge/sqlx/issues/3412#issuecomment-2277379818)

Example error message:

```
Diagnostics:
1. the trait bound `chrono::DateTime<chrono::Utc>: std::convert::From<tower_sessions::cookie::time::OffsetDateTime>` is not satisfied
   the following other types implement trait `std::convert::From<T>`:
     `chrono::DateTime<chrono::Utc>` implements `std::convert::From<chrono::DateTime<chrono::FixedOffset>>`
     `chrono::DateTime<chrono::Utc>` implements `std::convert::From<chrono::DateTime<chrono::Local>>`
     `chrono::DateTime<chrono::Utc>` implements `std::convert::From<std::time::SystemTime>`
   required for `tower_sessions::cookie::time::OffsetDateTime` to implement `std::convert::Into<chrono::DateTime<chrono::Utc>>` [E0277]
```

**Comment by maxcountryman** [comment]
Reading that issue, this isn't related to `tower-sessions`; the specific issue you're encountering is due to `tower-session-sqlx-store` enabling the time feature.

**Reply by vinnymeller** [reply]
@maxcountryman correct in that sense, but `time` has to be enabled for sqlx in `tower-session-sqlx-store` because `time::OffsetDateTime` is used in `Record`'s `expiry_date` and elsewhere, so you can't simply swap out chrono in place of time in `tower-session-sqlx-store` unless I'm missing something

**Reply by maxcountryman** [reply]
By that logic you should file this against `cookie` since that's ultimately what requires `time`. Regardless, it's not an issue in `tower-sessions`.
